### ISSUE TITLE
[vite] Internal server error: Dynamic require of "@svgr/core" is not supported

### ISSUE THREAD

**Issue by IanVS** [antfu/unplugin-icons#65]
I'm trying to use this package with vite and react, and I've followed the readme but I'm getting the following error during build:

```
[vite] Internal server error: Dynamic require of "@svgr/core" is not supported
```

This seems to be triggered from https://github.com/antfu/unplugin-icons/blob/038cf0e5431ff0f4889f5f150b18e74f19ebdced/src/core/compilers/jsx.ts#L6

Relevant parts of my vite.config.js:

```
import Icons from 'unplugin-icons/vite';
import { FileSystemIconLoader } from 'unplugin-icons/loaders';

export default defineConfig({
  plugins: [
    Icons({
      compiler: 'jsx',
      jsx: 'react',
      customCollections: {
        myproject: FileSystemIconLoader('./src/assets/icons'),
      },
    }),
  ],
});
```

**Comment by antfu** [comment]
Did you install `@svgr/core`? Can you share a minimal repro? Thanks

**Comment by IanVS** [comment]
I'm trying to reproduce in the `vite-react` example right now, but having troubles running that too.  I'll open a separate issue for it, though.  Maybe I'm just doing something wrong.

**Comment by IanVS** [comment]
This can be triggered by adding `"type": "module"` to the example's package.json file.

**Comment by antfu** [comment]
Hopefully fixed in v0.11.4

**Comment by IanVS** [comment]
Unfortunately this does not seem fixed for me, using the latest 0.12.13.  I believe it is because of the compiled code:

```
// src/core/compilers/jsx.ts
var JSXCompiler = async (svg, collection, icon, options) => {
  const svgr = (await Promise.resolve().then(() => __toModule(__require("@svgr/core")))).default;
  let res = await svgr(svg, {}, { componentName: camelize(`${collection}-${icon}`) });
  if (options.jsx !== "react")
    res = res.replace('import * as React from "react";', "");
  return res;
};
```

Maybe it works in the example now because it is using the `src` directly, and not the compiled build?
